test/integration/src/peggy2/test_eth_transfers.py

```diff
@@ -140,14 +144,276 @@ def transfer_erc20_to_sifnode_and_back(ctx, token_sc, token_decimals, number_of_
         assert eth_balance_after_1 == send_amount_1 * (i + 1)
 
 
+# Lock an eth to
+# We expect the transfer from evm to sifchain to be successful,
+# We expect the transfer from sifchain to evm to fail, w/
+#   - Tokens burned on sifchain side
+#   - Tokens not depsoited on evm side
+#   - It SHOULD NOT halt the bridge
+def test_failhard_token_to_sifnode_and_back(ctx: EnvCtx):
+    test_eth_acct = ctx.create_and_fund_eth_account(fund_amount=fund_amount_eth)
+    test_sif_account = ctx.create_sifchain_addr(fund_amounts=[[fund_amount_sif, "rowan"]])
+                                                            #   [fund_amount_ceth_cross_chain_fee, ctx.ceth_symbol]])
+    test_account_token_balance = 30000
+
+    token_sc = deploy_failhard_for_test(ctx, test_eth_acct, test_account_token_balance)
+    token_addr = token_sc.address
+    sif_denom_hash = sifchain.sifchain_denom_hash(ctx.ethereum_network_descriptor, token_sc.address)
+
+    ctx.bridge_bank_lock_eth(test_eth_acct, test_sif_account, 5000)
+    ctx.advance_blocks(100)
+
+    sleep(10)
```

Comment:
in the siftool environment, there is no wait to do a wait on a transaction?

Reply:
Actually, as I read more of the code, I am less sure what this sleep is for; we do transactions without sleeping. Why is this needed?

Reply:
This is required because we need to account for time for ebrelayer to pick up the block, and relay it to cosmos, and for cosmos to process it. However, this will be removed in this PR